hoot-core/src/main/cpp/hoot/core/io/OsmJsonReader.h

```diff
@@ -0,0 +1,225 @@
+ *   "   'id': -3,                           \n"
+ *   "   'lat': 4.0,                         \n"
+ *   "   'lon': -3.0,                        \n"
+ *   "   'tags': {                           \n"
+ *   "     'highway': 'bus_stop',            \n"
+ *   "     'name': 'Micah\\'s Street'        \n"
+ *   "   }                                   \n"
+ *   " }                                     \n"
+ *   "]                                      \n"
+ *   "}                                      \n";
+ *
+ * It's all-or-nothing, though for the quotes - don't mix and match
+ * singles and doubles! Also, be aware that this class doesn't do anything
+ * clever to handle large datasets - it simply keeps everything in memory.
+ * Be careful if you want to use it with large datasets.
+ */
+
+class OsmJsonReader
+{
+public:
+
+  // Default consructor
+  OsmJsonReader();
+
+  /**
+   * @brief loadFromString - Builds a map from the JSON string. Throws a
+   *        HootException with error and line number if JSON parsing fails
+   * @param jsonStr - input string
+   * @return Smart pointer to the OSM map
+   */
+  OsmMapPtr loadFromString(QString jsonStr);
+
+  /**
+   * @brief loadFromFile - Reads the whole file as a string, passes it
+   *        to loadFromString()
+   * @param path - Path to file
+   * @return Smart pointer to the OSM map
+   */
+  OsmMapPtr loadFromFile(QString path);
+
+  /**
+   * @brief getVersion Overpass API version, if that's where JSON comes from
+   * @return version string
+   */
+  QString getVersion() { return _version; }
+
+  /**
+   * @brief getGenerator Map generator name, if supplied in JSON
+   * @return generator name string
+   */
+  QString getGenerator() { return _generator; }
+
+  /**
+   * @brief getTimestampBase OSM timestamp base, if supplied in JSON
+   * @return timestamp base string
+   */
+  QString getTimestampBase() { return _timestamp_base; }
+
+  /**
+   * @brief getCopyright Copyright statement, if supplied in JSON
+   * @return copyright string
+   */
+  QString getCopyright() { return _copyright; }
+
+private:
+
+  /**
+   * @brief _loadJSON Loads JSON into a boost property tree
+   * @param jsonStr String to load
+   */
+  void _loadJSON(QString jsonStr);
+
+  /**
+   * @brief _parseOverpassJson Traverses our property tree to build
+   *        the OSM map object
+   * @return Pointer to the map we built
+   */
+  OsmMapPtr _parseOverpassJson();
+
+  // Our property tree that holds JSON
+  boost::property_tree::ptree _propTree;
+
+  // Store these items from overpass api
+  QString _version;
+  QString _generator;
+  QString _timestamp_base;
+  QString _copyright;
+
+  /**
+   * @brief parseOverpassNode Reads node info out of the property tree and
+   *        builds a Node object. Adds the node to the map.
+   * @param item Property Tree (likely a sub-tree)
+   * @param pMap Map to which we add the node
+   */
+  void parseOverpassNode(const boost::property_tree::ptree &item,
```

Comment:
While we don't have an official "code guide", we've been using "_" prefix on private/protected members. E.g. _parseOverpassNode.